### ISSUE TITLE
Release notes links point to module rather than section

### ISSUE THREAD

**Issue by bsipocz** [scipy/scipy#11620]
I would expected the links in the release notes to point to the section in the notes rather than to the module docs.

E.g. to https://docs.scipy.org/doc/scipy/reference/release.1.4.0.html#scipy-stats-improvements
rather than to https://docs.scipy.org/doc/scipy/reference/stats.html#module-scipy.stats

I'm happy to work on a fix if this is not intentional and others would also expect  the behaviour I had.

**Comment by rgommers** [comment]
This confused me for a little while. I now see what you mean:

<img width="1185" alt="image" src="https://user-images.githubusercontent.com/98330/75865600-14332e00-5e04-11ea-8ea7-8f3d3205b449.png">

The sidebar on the right has the links to the doc sections of the release notes, the ones on the left are to the refguide module overview pages. The reason for the latter is that they're the links from the section headers themselves:

<img width="823" alt="image" src="https://user-images.githubusercontent.com/98330/75865762-52c8e880-5e04-11ea-9191-ce85cada8df9.png">

Making those links point to themselves makes little sense though. So probably the right thing to do here is to remove the link from the section headers completely, and make the main ToC a normal Sphinx TOC - that will then have the same links as in the right sidebar.

**Comment by bsipocz** [comment]
oh, somehow I never noticed the sidebar, so much focused on the middle content 😅

**Comment by lucascolley** [comment]
> probably the right thing to do here is to remove the link from the section headers completely, and make the main ToC a normal Sphinx TOC - that will then have the same links as in the right sidebar.

@rgommers do you still want something done here? The contents entries still link to the module docs, but we no longer have the links per module in the right-sidebar.

**Comment by rgommers** [comment]
This looks reasonable to me now:

<img width="923" alt="image" src="https://github.com/scipy/scipy/assets/98330/91121b9e-1225-49ea-975f-1f7aae90e373">

I'd be fine with closing it.

**Comment by lucascolley** [comment]
Yeah, I agree with @bsipocz that it seems unintuitive for the links to go to the module ref docs, but probably only worth changing if there's something easy and clearly better. Any thoughts @bsipocz ?

**Comment by j-bowhay** [comment]
This looks like it is now fixed https://docs.scipy.org/doc/scipy/release/1.16.0-notes.html
